### PR TITLE
Add a channel buffer to decouple abci and grpc streaming

### PR DESCRIPTION
**Pull request by jonfung-dydx**
Adds a buffered go channel between the abci execution code and the grpc stream emission in order to not block abci execution code on grpc stream emission

## Summary by CodeRabbit

- **New Features**
  - Added support for configuring GRPC streaming buffer size.
  - Introduced a new method to stop GRPC streaming manager.
  - Enhanced logging to include buffer size information when GRPC streaming is enabled.

- **Bug Fixes**
  - Fixed an issue where the GRPC streaming manager was not properly stopped when a new instance was created.

- **Improvements**
  - Improved orderbook update handling by grouping updates and enqueuing them efficiently.
  - Enhanced test coverage for new GRPC streaming buffer size configuration.

- **Metrics**
  - Added new metrics for GRPC streaming buffer size and number of connections.

### CODE REVIEW

File: protocol/streaming/grpc/grpc_streaming_manager.go [hunk 1]
```diff
@@ -36,16 +53,78 @@ type OrderbookSubscription struct {
...
+	delete(sm.orderbookSubscriptions, id)
+}
+
+func (sm *GrpcStreamingManagerImpl) sendUpdateResponse(
+	internalResponse bufferInternalResponse,
+) {
+	// Send update to subscribers.
+	subscriptionIdsToRemove := make([]uint32, 0)
+
+	for id, subscription := range sm.orderbookSubscriptions {
```
Comment (jayy04) [inline protocol/streaming/grpc/grpc_streaming_manager.go:108]: do we need locking here? or sync.Map?

**Review by jayy04 — APPROVED** [review]

File: protocol/streaming/grpc/grpc_streaming_manager.go (hunk 1 above)
Comment (jonfung-dydx) [reply protocol/streaming/grpc/grpc_streaming_manager.go:108]: discussed in person. non blocking
https://linear.app/dydx/issue/CT-869/grpc-streaming-correctness-investigate-locking-or-syncmap-for

**Comment by jonfung-dydx** [comment]
@mergifyio backport v4.1.x

**Comment by jonfung-dydx** [comment]
https://github.com/Mergifyio backport release/protocol/v4.1.x

**Pull request merged**